apis/network/v1alpha3/types.go

```diff
@@ -254,15 +253,18 @@ type PublicIPAddressFormat struct {
 	// PublicIPAllocationMethod - The public IP address allocation method. Possible values include: 'Static', 'Dynamic'
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Enum=Static;Dynamic
+	// +immutable
 	PublicIPAllocationMethod string `json:"allocationMethod"`
 
-	// PublicIPAllocationMethod - The public IP address version. Possible values include: 'IPV4', 'IPV6'
+	// PublicIPAllocationMethod - The public IP address version. Possible values include: 'IPv4', 'IPv6'
 	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Enum=IPV4;IPV6
+	// +kubebuilder:validation:Enum=IPv4;IPv6
+	// +immutable
 	PublicIPAddressVersion string `json:"version"`
 
 	// Location - Resource location.
 	// +optional
```

Comment:
Is this really optional?

Reply:
Turns out not to be. Marked it as required.